### ISSUE TITLE
Record position as RINEX

### ISSUE THREAD

**Issue by spono** [opengisch/QField#5381]
Hi,
I post it here as a possible idea for a future enhancement /plugin.

Roaming a bit looking for a way to record RINEX data from the device’s internal GPS I found this project (recent but I don’t know if it’s still active). I’m not a developer and I don’t have the knowledge base to understand but -I think- it may provide a good base for integration of such tools to the "average position" recording option that has been included in QField in the last releases.

https://github.com/luiscancillo/RINEXtools

HTH

**Comment by APOS80** [comment]
You need a Gnss rover that can record it, simple gnss receivers like in phones can’t.

**Comment by gdt** [comment]
I don't think this enhancement request makes sense.  RINEX is for recording carrier phase data.  QField expects to get positions.  Yes, one could take proprietary format streams from a receiver and transform to RINEX and then later postprocess to get positions -- but what does that have to do with QField?

One could create an app that e.g. wraps RTKLIB and computes RTK position and sends them to QField (by mocking, or some other interface), but that's something else too.

I recommend closing this issue.

**Comment by edgecase14** [comment]
> I don't think this enhancement request makes sense. RINEX is for recording carrier phase data. QField expects to get positions. Yes, one could take proprietary format streams from a receiver and transform to RINEX and then later postprocess to get positions -- but what does that have to do with QField?

Trimble has a feature where you can post-process to get high accuracy in the absence of RTK (radio link down, too far from RTK base, etc.)  by recording raw observations (carrier phase data) for a short period where you would normally collect a position solution.

So I think the request may make sense.

**Comment by gdt** [comment]
You haven't explained the data flow and how it would be used.   QField is very much about showing your current position as a cursor for visualizing, navigating, and digitizing a feature locked to the position.

You seem to envision some scheme where carrier phase data is recorded instead in a geometry field -- something for which there is no support in geopackages or the OGC schemas -- and then thinking that someone is going to write or adapt open source code to fetch reference station data and do PPK and convert that RINEX into a position.

There could be such a feature, but it's really big, involving not only QField but modifying OGC standards, geopackage, gdal, and needs an articulation of how it would behave, what the processing chain would be, and why it makes sense.  Maybe you can submit a PR for review and comments.

**Comment by edgecase14** [comment]
OK perhaps I shouldn't have commented in a way that might get anyone's hopes up.  It's certainly at the bottom of my list of priorities.

That said, a minimal implementation might be similar to how photos are collected.  A position is collected at "standard" accuracy, and a binary file capture of raw observations (collected for a minute?) could be "attached".  Beyond that it would be up to some very curious individual to post-process the rest and integrate the higher accuracy positions back into QGIS.  Someone who wants to fiddle around for an hour per point.  More of a "for the record" sort of comment than any sort of endorsement that effort should be spent on it.

**Comment by gdt** [comment]
Something would be possible.    I just meant that it sitting open as a ticket, esp given the project's push towards their separate feature tracker, doesn't seem for the best.    Someone needs to do a grand design for the concept in general and publish the design, and then prototype.  I'd cheer you on and review if you do that!

**Comment by edgecase14** [comment]
Not for this feature, but for a few others I've got in my queue, what is the community's way to publish designs and discuss?

**Comment by gdt** [comment]
I am not part of upstream.  Maybe someone who is will comment.

If it's clear enough that what you're doing is sensbile and there are no difficult architectural issues, and you've written a code, a PR seems fine.  Like an NTRIP client that stuffs the RTCM3 over bluetooth SPP or BLE or TCP.  It's great to actually write the code, but it's not breaking new ground conceptually.

If you need to write an architecture paper explaining why, the protocol changes, and a sketch of software changes to N packages (such as for this ticket), I'd post that on your website and write to the qgis dev list, and open up a feature discussion using the tool that qfield points to and link it.  Also post on the other packages you need to modify, qgis, gdal, postgis, and maybe some ogc standards group.
